src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Directives.cs

```diff
@@ -280,6 +291,8 @@ private void TagHelperDirective(string keyword, Func<string, ISpanCodeGenerator>
 
             var foundWhitespace = At(CSharpSymbolType.WhiteSpace);
             AcceptWhile(CSharpSymbolType.WhiteSpace);
+            var keywordStartLocation = Span.Start;
```

Comment:
Don't need to do this here. Just grab `CurrentLocation` prior to accepting the directive name. Doing `Span.Start` is a worse case :smile: